internal/otelcontextmapping_linux.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+//go:build linux
+
+// This is a go port of https://github.com/DataDog/fullhost-code-hotspots-wip/blob/main/lang-exp/anonmapping-clib/otel_process_ctx.c
+
+package internal
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"sync/atomic"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// These two constants are not in x/sys/unix by default; copy them from <linux/prctl.h>.
+	//nolint:revive
+	PR_SET_VMA = 0x53564D41
+	//nolint:revive
+	PR_SET_VMA_ANON_NAME = 0
+
+	otelContextSignature = "OTEL_CTX"
+)
+
+var (
+	otelContextMappingSize = 2 * os.Getpagesize()
+
+	existingMappingBytes []byte
+)
+
+type processContextHeader struct {
```

Comment:
Let's add a [`_ structs.HostLayout`](https://pkg.go.dev/structs#HostLayout) field to the beginning of this struct. It seems to work fine without it, but adding that field would ensure that the memory layout matches the C layout, which is what I gather the consumer of this data will expect. Especially since we copy the raw memory for this struct below. Makes it more future-proof.